node/src/components/contract_runtime/rewards.rs

```diff
@@ -0,0 +1,380 @@
+    pub async fn populate<REv: ReactorEventT>(
+        effect_builder: EffectBuilder<REv>,
+        protocol_version: ProtocolVersion,
+        block_hashs: impl IntoIterator<Item = (EraId, BlockHash)>,
+    ) -> Result<Self, PopulateError> {
+        let eras_info = stream::iter(block_hashs)
+            .then(|(era_id, block_hash)| async move {
+                let state_root_hash = effect_builder
+                    .get_block_from_storage(block_hash)
+                    .await
+                    .map(|block| *block.state_root_hash())
+                    .ok_or_else(|| PopulateError::FailedToFetchBlock(block_hash))?;
+                let weights = effect_builder
+                    .get_era_validators_from_contract_runtime(EraValidatorsRequest::new(
+                        state_root_hash,
+                        protocol_version,
+                    ))
+                    .await
+                    .map_err(PopulateError::FailedToFetchEra)?
+                    .into_values()
+                    .next()
+                    .ok_or_else(|| PopulateError::NoEraReturned(state_root_hash))?;
+                let total_supply = effect_builder
+                    .get_total_supply(state_root_hash)
+                    .await
+                    .map_err(PopulateError::FailedToFetchTotalSupply)?;
+                let seignorate_rate = effect_builder
+                    .get_round_seigniorage_rate(state_root_hash)
+                    .await
+                    .map_err(PopulateError::FailedToFetchSeigniorageRate)?;
+                let reward_per_round = seignorate_rate * total_supply;
+                let total_weights = weights.values().copied().sum();
+
+                Ok((
+                    era_id,
+                    EraInfo {
+                        weights,
+                        total_weights,
+                        reward_per_round,
+                    },
+                ))
+            })
+            .try_collect()
+            .await?;
+        let eras_info = ErasInfo(eras_info);
+
+        Ok(eras_info)
+    }
+
+    /// Returns the validators from a given era.
+    pub fn validator_keys(
+        &self,
+        era_id: EraId,
+    ) -> Result<
+        Either<impl Iterator<Item = PublicKey> + '_, impl Iterator<Item = PublicKey> + '_>,
+        RewardsError,
+    > {
+        if era_id.is_genesis() {
+            return Ok(Either::Left(std::iter::empty()));
+        }
+
+        let keys = self
+            .0
+            .get(&era_id)
+            .ok_or(RewardsError::EraIdNotInEraRange(era_id))?
+            .weights
+            .keys()
+            .cloned();
+
+        Ok(Either::Right(keys))
+    }
+
+    /// Returns the total potential reward pot per block.
+    /// Since it is per block, we do not care about the expected number of blocks per era.
+    pub fn reward_pot(&self, era_id: EraId) -> Result<Ratio<U512>, RewardsError> {
```

Comment:
If it's returning the round reward, then it should just be called "reward." Calling it a "reward pot" confuses the internal discussion where we use "reward pot" to refer to the entire per-era potential payout (i.e., reward * expected number of blocks).

Reply:
Done